app.js

```diff
@@ -134,14 +151,26 @@ router.post('/account/recoveryMethods', checkAccountOwnership, async ctx => {
     ctx.body = await recoveryMethodsFor(account);
 });
 
-async function withAccount(ctx, next) {
-    const { accountId } = ctx.request.body;
-    ctx.account = await models.Account.findOne({ where: { accountId } });
-    if (ctx.account) {
-        await next();
-        return;
+function getWithSequelizeAccountHandler(source) {
```

Comment:
nit: I think it's pretty common to have middleware with arguments (e.g. `ratelimit` middleware), so it makes sense to call outer function `withSequelizeAccount` instead of `getWithSequelizeAccountHandler`.

Reply:
I see that the `koa-ratelimit` middleware shadowed the same named function for the actual middleware function as it did for the middleware creator, but I'm not convinced that it is a good pattern to mimic generally.  These two functions do two completely different things, and run at different points in the lifecycle of the app.  One is a synchronous function that is run only once, during app setup where the other is an async method that is executed with every request that comes in.  Also, a key difference between this and the `koa-ratelimit` implementation is that this thunk actually does validation synchronously, during creation of the middleware, so there is an expectation that an error can be  thrown from the middleware _creator_ function; I generally especially try to avoid shadowing functions with identical names in cases like this where errors can be thrown from either 'layer' of the thunk, even apart from the other issues I mentioned above